src/borgstore/backends/sftp.py

```diff
@@ -40,7 +40,20 @@ def __init__(self, hostname: str, path: str, port: int = 22, username: Optional[
     def _connect(self):
         ssh = paramiko.SSHClient()
         ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        ssh.connect(hostname=self.hostname, username=self.username, port=self.port, allow_agent=True)
+        config_path = Path("~/.ssh/config").expanduser()
+        try:
+            sc = paramiko.SSHConfig.from_path(str(config_path))
+        except FileNotFoundError:
+            scd = paramiko.SSHConfigDict()  # empty dict
+        else:
+            scd = sc.lookup(self.hostname)
+        ssh.connect(
+            hostname=scd.get("hostname", self.hostname),
+            username=scd.get("user", self.username),
```

Comment:
Also take username from URL with precedence if provided

Reply:
Somehow my rebase onto current master has gone very wrong and lost 4 commits.

But I still have the stuff on my other dev machine, so I'll force push that soon.

Reply:
OK, fixed that. Just squashed all past commits together.